model/src/test/kotlin/utils/LicenseResolverTest.kt

```diff
@@ -0,0 +1,244 @@
+
+import org.ossreviewtoolkit.model.AccessStatistics
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.CopyrightFinding
+import org.ossreviewtoolkit.model.Environment
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.LicenseFinding
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Repository
+import org.ossreviewtoolkit.model.ScanRecord
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScanResultContainer
+import org.ossreviewtoolkit.model.ScanSummary
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.ScannerRun
+import org.ossreviewtoolkit.model.TextLocation
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicInteger
+
+private fun OrtResult.addPackage(pkg: Package): OrtResult = copy(
+    analyzer = analyzer!!.copy(
+        result = analyzer!!.result.copy(
+            packages = (analyzer!!.result.packages + pkg.toCuratedPackage()).toSortedSet()
+        )
+    )
+)
+
+private fun OrtResult.addScanResult(id: Identifier, scanResult: ScanResult): OrtResult {
+    val scanResults = scanner!!.results.scanResults
+    var scanResultsForId = scanResults.find { it.id == id }?.let {
+        scanResults.remove(it)
+        it.results
+    } ?: emptyList()
+
+    scanResults += ScanResultContainer(id, scanResultsForId + scanResult)
+
+    return copy(
+        scanner = scanner!!.copy(
+            results = scanner!!.results.copy(
+                scanResults = scanResults
+            )
+        )
+    )
+}
+
+private fun OrtResult.getVcsProvenance(id: Identifier): Provenance? =
+    getProject(id)?.let { Provenance(vcsInfo = it.vcsProcessed) }
+        ?: getPackage(id)?.let { Provenance(vcsInfo = it.pkg.vcsProcessed) }
+
+private fun OrtResult.getSourceArtifactProvenance(id: Identifier): Provenance? =
+    getPackage(id)?.let { Provenance(sourceArtifact = it.pkg.sourceArtifact) }
+
+private fun createEmptyOrtResult() = OrtResult(
+    Repository.EMPTY,
+    AnalyzerRun(
+        environment = Environment(),
+        config = AnalyzerConfiguration(ignoreToolVersions = true, allowDynamicVersions = true),
+        result = AnalyzerResult.EMPTY
+    ),
+    ScannerRun(
+        environment = Environment(),
+        config = ScannerConfiguration(),
+        results = ScanRecord(
+            scanResults = sortedSetOf(),
+            storageStats = AccessStatistics()
+        )
+    )
+)
+
+class LicenseResolverTest : WordSpec() {
+    private val nextId = AtomicInteger(1)
```

Comment:
What's the reason for this being an `AtomicInteger` instead of a regular `Int`, as there seems to be no concurrency involved here?

Reply:
The reasons are that one can use `val` instead of `var` and that I found `getAndIncrement()` more readable mainly.